### PR TITLE
add documentation for .refineStateType

### PR DESCRIPTION
**Pull request by Kelerchian**

### CODE REVIEW

File: machine-runner/README.md [hunk 1]
```diff
@@ -327,6 +327,51 @@ const stateAsInitialWarehouse = state
...
+    // here the type of payload is narrowed to { robot: string }
+  }
+}
+```
+
+The argument to `.refineStateType` must be an array containing all previously defined states.
+Any other argument will throw an error.
+
+The aliasing machine shares the original machine's internal state.
+All method calls, such as `.destroy`, create the same effect as when enacted on the original machine.
```
Comment (jmg-duarte) [inline machine-runner/README.md:373]: This makes me wonder if the machine is still the same or a copy.

Sure it says "aliasing" (shouldn't it be "aliased"?) but "create the same effect as when enacted on the original machine" makes the line blurry.

File: machine-runner/README.md [hunk 2]
```diff
@@ -327,6 +327,51 @@ const stateAsInitialWarehouse = state
   .request(from: `source`, to: `destination`);
 ```
 
+### `refineStateType`
+
+A MachineRunner instance now has a new method available: `refineStateType` which return a new aliasing machine.
```
Comment (jmg-duarte) [inline machine-runner/README.md:332]:
```suggestion
A `MachineRunner` instance now has a new method available: `refineStateType` which return a new aliasing machine.
```

File: machine-runner/README.md (hunk 1 above)
Comment (Kelerchian) [reply machine-runner/README.md:373]: It's copy of **only** the "shell". I'm not sure if the word should be "aliased" or "aliasing". But I assume, aliased is the object being aliased, aliasing is the object that aliases other object.

Comment (Kelerchian) [reply machine-runner/README.md:373]: Yes. Sorry for the late response

**Review by rkuhn — APPROVED** [review]
LGTM

**Pull request merged**